### PR TITLE
Add compactor start duration in seconds metric

### PR DESCRIPTION
**Pull request by alexqyle**
**What this PR does**:
Add additional metrics to record compactor start and stop duration in seconds. These metrics would be helpful to keep track of abnormally long start or stop time of compactor.

**Which issue(s) this PR fixes**:
NA

**Checklist**
- [ ] Tests updated
- [ ] Documentation added
- [ ] `CHANGELOG.md` updated - the order of entries should be `[CHANGE]`, `[FEATURE]`, `[ENHANCEMENT]`, `[BUGFIX]`

### CODE REVIEW

File: pkg/compactor/compactor.go [hunk 1]
```diff
@@ -581,6 +596,11 @@ func (c *Compactor) starting(ctx context.Context) error {
 }
 
 func (c *Compactor) stopping(_ error) error {
+	begin := time.Now()
+	defer func() {
+		c.CompactorStopDurationSeconds.Set(time.Since(begin).Seconds())
```
Comment (yeya24) [inline pkg/compactor/compactor.go:601]: After emitting the metric, compactor process stops immediately.
Then how can Prometheus scrapes it?

File: CHANGELOG.md [hunk 2]
```diff
@@ -6,6 +6,7 @@
 * [CHANGE] Index Cache: Multi level cache backfilling operation becomes async. Added `-blocks-storage.bucket-store.index-cache.multilevel.max-async-concurrency` and `-blocks-storage.bucket-store.index-cache.multilevel.max-async-buffer-size` configs and metric `cortex_store_multilevel_index_cache_backfill_dropped_items_total` for number of dropped items. #5661
 * [FEATURE] Ingester: Add per-tenant new metric `cortex_ingester_tsdb_data_replay_duration_seconds`. #5477
 * [ENHANCEMENT] Store Gateway: Added `-store-gateway.enabled-tenants` and `-store-gateway.disabled-tenants` to explicitly enable or disable store-gateway for specific tenants. #5638
+* [ENHANCEMENT] Compactor: Add compactor start and stop duration in seconds metrics. #5683
```
Comment (yeya24) [inline CHANGELOG.md:9]: Let's mention the new metric name here

File: pkg/compactor/compactor.go (hunk 1 above)
Comment (wenxu1024) [reply pkg/compactor/compactor.go:601]: Could we add a finalSleep similar to the lifecyler? So we add a finalSleep here to make sure the stopDuration metrics are collected.

https://github.com/cortexproject/cortex/blob/master/pkg/ring/lifecycler.go#L895

Comment (yeya24) [reply pkg/compactor/compactor.go:601]: I don't think it makes sense to add a new flag of sleep time in order to catch a metric.

Comment (alexqyle) [reply pkg/compactor/compactor.go]: Discussed offline, using log instead of metric for stop duration

Comment (yeya24) [reply pkg/compactor/compactor.go]: @wenxu1024 seems you are right. The final sleep is already available in code and we just need to expose it via compactor flag.
But Idk if it is worth to wait a longer time when stopping in order to capture the metric.

Comment (wenxu1024) [reply pkg/compactor/compactor.go]: got it. Seems to collect this new metric, we need to sleep in this stopping function (not in the lifecycler). A log would be sufficient for now.

**Pull request merged**